R/setkey.R

```diff
@@ -155,16 +155,24 @@ setreordervec = function(x, order) .Call(Creorder, x, order)
 # The others (order, sort.int etc) are turned off to protect ourselves from using them internally, for speed and for
 # consistency; e.g., consistent twiddling of numeric/integer64, NA at the beginning of integer, locale ordering of character vectors.
 
-is.sorted = function(x, by=seq_along(x)) {
+is.sorted = function(x, by=seq_along(x), retOrd=FALSE) {
   if (is.list(x)) {
-    warning("Use 'if (length(o <- forderv(DT,by))) ...' for efficiency in one step, so you have o as well if not sorted.")
+    # for efficient use via retOrd argument see note in ?is.sorted
```

Comment:
What is the use case of ```is.sorted(..., retOrd = TRUE)```? The order seems better dealt with exporting ```forder```.

Reply:
Agree, use case address the warning without exporting forder